### PR TITLE
Rover: shorten radio and GCS failsafe timeouts

### PR DESCRIPTION
**Pull request by rmackay9**
This PR improves the Rover failsafe in a few ways brought up in this issue https://github.com/ArduPilot/ardupilot/issues/6623:

- reduce default failsafe timeouts for radio and GCS failsafes.  Note the timeout for each includes a hardcoded 1 or 2 seconds plus the value held in FS_TIMEOUT meaning the real default timeouts are:
  - radio/throttle failsafe default timeout = 2.5sec (1sec + FS_TIMEOUT)
  - GCS failsafe default timeout = 3.5 sec (2sec + FS_TIMEOUT)
- GCS failsafe will not trigger if RC_OVERRIDE or MANUAL_CONTROL messages are arriving.  This helps reduce false positives when a joystick is being used which may use up much of the bandwidth and lead to HEARTBEAT messages arriving more slowly than 1hz.
- non-functional parameter description updates, renamed a function, moved a variable to a structure, etc

This PR does not attempt to solve all the problems in the Rover failsafe code.  It's just some small improvements that came out of an investigation into [this Mission Planner issue](https://github.com/ArduPilot/MissionPlanner/issues/1972).

### CODE REVIEW

**Comment by rmackay9** [comment]
@jmachuca77 is planning to test this on a real vehicle for me I think.

File: APMrover2/GCS_Mavlink.cpp [hunk 1]
```diff
@@ -818,18 +820,18 @@ void GCS_MAVLINK_Rover::handleMessage(mavlink_message_t* msg)
...
         {
-            // We keep track of the last time we received a heartbeat from our GCS for failsafe purposes
+            // we keep track of the last time we received a heartbeat from our GCS for failsafe purposes
             if (msg->sysid != rover.g.sysid_my_gcs) {
                 break;
             }
-
-            rover.last_heartbeat_ms = AP_HAL::millis();
-            rover.failsafe_trigger(FAILSAFE_EVENT_GCS, false);
+            rover.failsafe.last_heartbeat_ms = AP_HAL::millis();
```
Comment (jmachuca77) [inline APMrover2/GCS_Mavlink.cpp]: Why is this setting it to AP_HAL::millis and the rest do it to tnow?

File: APMrover2/APMrover2.cpp [hunk 2]
```diff
@@ -195,9 +195,13 @@ void Rover::ahrs_update()
...
 {
-    if (g.fs_gcs_enabled) {
-        failsafe_trigger(FAILSAFE_EVENT_GCS, last_heartbeat_ms != 0 && (millis() - last_heartbeat_ms) > 2000);
+    if (!g.fs_gcs_enabled) {
+        // gcs failsafe disabled
+        return;
     }
+
+    // check for updates from GCS within 2 seconds
+    failsafe_trigger(FAILSAFE_EVENT_GCS, failsafe.last_heartbeat_ms != 0 && (millis() - failsafe.last_heartbeat_ms) > 2000);
```
Comment (jmachuca77) [inline APMrover2/APMrover2.cpp]: This still has the hardcoded 2 second timeout.

Comment (rmackay9) [reply APMrover2/APMrover2.cpp]: In the case you're testing though it's the time of the throttle failsafe that is important.  That's been reduced from 2 seconds to 1 second (+FS_TIMEOUT value).  We can potentially make it shorter still though.

File: APMrover2/GCS_Mavlink.cpp (hunk 1 above)
Comment (rmackay9) [reply APMrover2/GCS_Mavlink.cpp]: tnow is set to millis() higher up.

Comment (jmachuca77) [reply APMrover2/GCS_Mavlink.cpp]: I saw that, but I think for consistency should it be set to the same as the others? Or set the others to AP_HAL::millis perhaps?

**Comment by rmackay9** [comment]
Merged after testing by @jmachuca77

**Comment by jmachuca77** [comment]
Tested on a rover, with a 200ms trigger for the Thr failsafe with good results. Good to go from here.

**Pull request merged**